sdk/core/Azure.Core/src/Serialization/SerializableOptions.cs

```diff
@@ -23,9 +26,14 @@ public class SerializableOptions
         /// </summary>
         public bool PrettyPrint { get; set; }
 
+        ///// <summary>
+        ///// todo
+        ///// </summary>
+        //public ObjectSerializer? Serializer { get; set; }
+
         /// <summary>
-        /// todo
+        /// TODO
         /// </summary>
-        public ObjectSerializer? Serializer { get; set; }
+        public Dictionary<Type, ObjectSerializer> Serializers = new Dictionary<Type, ObjectSerializer>();
```

Comment:
Should we mark this as `readonly` or add  `{ get; }`?

Reply:
I think this cannot be readonly as the user needs to be able to add model types and their associated serializers.

Reply:
They would still be able to add elements to the dictionary - just not replace the dictionary instance.

Reply:
sorry Chris I misunderstood you yesterday - updating now.